addons/website_slides/static/src/js/slides_upload.js

```diff
@@ -31,23 +34,26 @@ var SlideUploadDialog = Dialog.extend({
             size: 'medium',
         });
         this._super(parent, options);
-        this._setup();
-
         this.channelID = parseInt(options.channelId, 10);
         this.defaultCategoryID = parseInt(options.categoryId,10);
         this.canUpload = options.canUpload === 'True';
         this.canPublish = options.canPublish === 'True';
         this.modulesToInstall = options.modulesToInstall ? JSON.parse(options.modulesToInstall.replace(/'/g, '"')) : null;
         this.modulesToInstallStatus = null;
-
+        this.removeButtons = false;
         this.set('state', '_select');
         this.on('change:state', this, this._onChangeType);
         this.set('can_submit_form', false);
         this.on('change:can_submit_form', this, this._onChangeCanSubmitForm);
-
         this.file = {};
         this.isValidUrl = true;
     },
+    willStart: function () {
+        var self = this;
+        return this._setup().then(
+            self._super.apply(this, arguments)
+        );
+    },
```

Comment:
no need to do make promise here.
if you want to do setup before will start, just do this._setup() as setup is returning only data. (without asynchronous brols)
but is better to do setup in init then.

Reply:
but I think there are asynchronous brols in _setup(), as in the override in slides_upload.js in website_slides_survey, I do an rpc call that comes after _super call,  that must finish before we choose what template we will use and this must finish before we render the template. therefore I made setup return a Promise. But i'm not sur that is how it must be done.

Reply:
If you want to wait for all deferred before start, you can do this, it will just do both simultaneously, but wait for both to finish : 
```
willStart: function () {
    var def = this._setup();
    return Promise.all([this._super.apply(this, arguments), def]);
},
```
or if you want to have setup(); done after super, you can do this : 
```
willStart: function () {
    var self = this;
    return this._super.apply(this, arguments).then(function () {
        return this._setup();
    });
},
```

But anyway, the warning could be in d-none in the template (de base) and you show the warning only if the dropdown list of available survey is empty (and if you cannot create). during the fill data (as you did in fact..) No need to load another templates, the user will do go back or close the form and that's it.